src/Chapter2.hs

```diff
@@ -620,7 +631,13 @@ Write a function that takes elements of a list only on even positions.
 >>> takeEven [2, 1, 3, 5, 4]
 [2,3,4]
 -}
-takeEven = error "takeEven: Not implemented!"
+takeEven :: [a] -> [a]
+takeEven lst = go (0::Int) lst
+  where
+    go _ [] = []
+    go n (x:xs)
+      | n `mod` 2 == 0 = x : go (n+1) xs
+      | otherwise      = go (n+1) xs
```

Comment:
Nice idea on the helper functions :+1:

A few additional thoughts on your idea: as you have only two states of the operation (odd or even), instead of accumulators of the current position number, you can just use `Bool`. And on each next recursion call, you just give it `not acc`. This will remove the necessity to check `mod 2` each time, which would save a couple of functions call on each interaction.

Alternatively, you could use pattern matching in `takeEven` instead. As you need to take every second element, it is possible to represent with a few pattern matching clauses 🙂 And it could be a bit faster 🙂

Reply:
I used pattern matching in `takeEven`. I'm not sure about how readable it is.  It might win at code golf, but it takes me a second to grok.  What do you think?

Reply:
I think it is a matter of habits. For me, the pattern matching version is more readable, as I immediately see the patterns that it looks for. 
But I agree that you need to write in your codebases what you think is more readable and understandable. 
The way you currently wrote the function probably would also take some time to understand, as you use `Int` as the representer of the state. However, both of the solutions are correct. The pattern-matching one would be more efficient by time and especially memory 🙂